action.go

```diff
@@ -182,11 +247,32 @@ func importActionV1(source map[string]interface{}) (*action, error) {
 		"started":   schema.Omit,
 		"completed": schema.Omit,
 	}
+	return fields, defaults
+}
+
+func actionV2Fields() (schema.Fields, schema.Defaults) {
+	fields, defaults := actionV1Fields()
+	fields["logs"] = schema.StringMap(schema.Any())
+	defaults["logs"] = schema.Omit
+	return fields, defaults
+}
+
+func importActionV1(source map[string]interface{}) (*action, error) {
+	fields, defaults := actionV1Fields()
+	return importAction(fields, defaults, 1, source)
+}
+
+func importActionV2(source map[string]interface{}) (*action, error) {
+	fields, defaults := actionV2Fields()
+	return importAction(fields, defaults, 2, source)
+}
+
```

Comment:
If you converted actions to use a map of fieldsFuncs then you wouldn't need this unsightly duplication. :)
Examples are cloudinstance or subnets.

Reply:
done